gateway/ln-gateway/src/lib.rs

```diff
@@ -464,35 +464,51 @@ impl Gateway {
         }
     }
 
+    async fn fetch_lightning_route_info_try(
+        lnrpc: &dyn ILnRpcClient,
+    ) -> Result<(Vec<RouteHint>, PublicKey, String)> {
+        let route_hints: Vec<RouteHint> = lnrpc
+            .routehints()
+            .await?
+            .try_into()
+            .expect("Could not parse route hints");
+
+        let GetNodeInfoResponse { pub_key, alias } = lnrpc.info().await?;
+        let node_pub_key = PublicKey::from_slice(&pub_key)
+            .map_err(|e| GatewayError::InvalidMetadata(format!("Invalid node pubkey {e}")))?;
+
+        Ok((route_hints, node_pub_key, alias))
+    }
+
     async fn fetch_lightning_route_info(
         lnrpc: Arc<dyn ILnRpcClient>,
     ) -> Result<(Vec<RouteHint>, PublicKey, String)> {
-        let mut num_retries = 0;
-        let (route_hints, node_pub_key, alias) = loop {
-            let route_hints: Vec<RouteHint> = lnrpc
-                .routehints()
-                .await?
-                .try_into()
-                .expect("Could not parse route hints");
-
-            let GetNodeInfoResponse { pub_key, alias } = lnrpc.info().await?;
-            let node_pub_key = PublicKey::from_slice(&pub_key)
-                .map_err(|e| GatewayError::InvalidMetadata(format!("Invalid node pubkey {e}")))?;
-
+        for num_retries in 0.. {
+            let (route_hints, node_pub_key, alias) =
+                match Self::fetch_lightning_route_info_try(lnrpc.as_ref()).await {
+                    Ok(res) => res,
+                    Err(e) => {
+                        if num_retries == ROUTE_HINT_RETRIES {
+                            return Err(e);
+                        }
+                        warn!("Could not fetch route hints: {e}");
```

Comment:
nit: I would have probably included the fact that it will be retried in `ROUTE_HINT_RETRY_SLEEP`